src/python_testing/TC_DGWIFI_2_3.py

```diff
@@ -0,0 +1,160 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+#
+
+import asyncio
+
+import chip.clusters as Clusters
+from chip.clusters.Types import Nullable, NullValue
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_DGWIFI_2_3(MatterBaseTest):
+    """
+    [TC-DGWIFI-2.3] Wi-Fi Network Diagnostics Cluster - Command Verification
+
+    This test case verifies the ResetCounts command and subsequent behavior.
+    """
+
+    @staticmethod
+    def is_valid_uint32(value):
+        """Validates a uint32 value."""
+        return isinstance(value, int) and 0 <= value <= 0xFFFFFFFF
+
+    async def send_reset_counts_command(self, endpoint):
+        """Sends the ResetCounts command to the DUT."""
+        cluster = Clusters.Objects.WiFiNetworkDiagnostics
+        await self.send_single_cmd(cluster.Commands.ResetCounts(), endpoint=endpoint)
+
+    async def read_attribute_and_validate(self, endpoint, attribute, validation_func, field_name):
+        """Reads an attribute and validates it using the provided function."""
+        value = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=Clusters.Objects.WiFiNetworkDiagnostics, attribute=attribute)
+        if value is not None:
+            asserts.assert_true(isinstance(value, Nullable), f"{field_name} must be of type 'Nullable' when not None.")
+            if value != NullValue:
+                asserts.assert_true(validation_func(value.Value), f"{field_name} has an invalid value: {value.Value}")
```

Comment:
consider decreasing indent sizes:

```suggestion
        if value is None:
            return
        asserts.assert_true(isinstance(value, Nullable), f"{field_name} must be of type 'Nullable' when not None.")
        if value == NullValue:
            return
        asserts.assert_true(validation_func(value.Value), f"{field_name} has an invalid value: {value.Value}")
```

Reply:
I think the CI will complain we skip assert, let me give a try